src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,17 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (!is_scalar($tagValue)
+                            && !(is_object($tagValue) && method_exists($tagValue, '__toString'))
```

Comment:
is it really up to the SpanOptions to force scalar values for types or cannot the implementation of tracer decide for itself? or do other implementations also have a requirement `map[string]scalar` of sorts?